frontend/src/metabase/querying/components/FilterModal/FilterModal.tsx

```diff
@@ -53,22 +54,25 @@ export const useFilterContent = (
   const handleChange = (newQuery: Lib.Query) => {
     setQuery(newQuery);
     setIsChanged(true);
+    // for handleSubmit to see the latest query if it is called in the same tick
```

Comment:
The "tick" is a bit misleading as it suggests JS stuff. It's actually React batching, right?